src/plugins/auto/executable_network.cpp

```diff
@@ -191,6 +192,33 @@ MultiDeviceExecutableNetwork::MultiDeviceExecutableNetwork(const std::string&
         std::list<DeviceInformation> validDevices =
             _multiPlugin->GetValidDevice(metaDevices, _loadContext[ACTUALDEVICE].networkPrecision);
 
+        // for the case of -d "AUTO"
+        if (context.noDevicePriority) {
+            std::list<DeviceInformation>::iterator itCPUDevice;
+            int GPUNums = 0, CPUNums = 0;
+            for (auto it = validDevices.begin(); it != validDevices.end(); it++) {
```

Comment:
better use range based loop code style

Reply:
may be can we infer enableDevicePriority directly from validDevices? instead of introducing new members of context?